sys/net/application_layer/dhcpv6/client.c

```diff
@@ -600,10 +773,85 @@ static void _parse_advertise(uint8_t *adv, size_t len)
     return;
 }
 
+static bool _parse_ia_na_option(dhcpv6_opt_ia_na_t *ia_na)
+{
+    if (!IS_USED(MODULE_DHCPV6_CLIENT_IA_NA)) {
+        return true;
+    }
+
+    for (unsigned i = 0; (i < CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX); i++) {
+        dhcpv6_opt_iaaddr_t *iaaddr = NULL;
+        addr_lease_t *lease = &addr_leases[i];
+        unsigned na_t1, na_t2;
+        uint32_t ia_id = byteorder_ntohl(ia_na->ia_id);
+        size_t ia_na_len = byteorder_ntohs(ia_na->len) -
+                            (sizeof(dhcpv6_opt_ia_na_t) - sizeof(dhcpv6_opt_t));
+        size_t ia_na_orig_len = ia_na_len;
+
+        if (lease->parent.ia_id.id != ia_id) {
+            continue;
+        }
+        /* check for status */
+        for (dhcpv6_opt_t *ia_na_opt = (dhcpv6_opt_t *)(ia_na + 1);
+                ia_na_len > 0;
+                ia_na_len -= _opt_len(ia_na_opt),
+                ia_na_opt = _opt_next(ia_na_opt)) {
+            if (ia_na_len > ia_na_orig_len) {
+                DEBUG("DHCPv6 client: IA_NA options overflow option "
+                        "boundaries\n");
+                return false;
+            }
+            switch (byteorder_ntohs(ia_na_opt->type)) {
+                case DHCPV6_OPT_STATUS: {
+                    if (!_check_status_opt((dhcpv6_opt_status_t *)ia_na_opt)) {
+                        continue;
+                    }
+                    break;
+                }
+                case DHCPV6_OPT_IAADDR: {
+                    dhcpv6_opt_iaaddr_t *this_iaaddr = (dhcpv6_opt_iaaddr_t *)ia_na_opt;
+                    if ((!lease->leased) ||
+                        (iaaddr == NULL)) {
+                        /* only take first address for now */
+                        iaaddr = this_iaaddr;
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+        na_t1 = byteorder_ntohl(ia_na->t1);
+        na_t2 = byteorder_ntohl(ia_na->t2);
+        _update_t1_t2(na_t1, na_t2);
+        if ((iaaddr != NULL)) {
```

Comment:
Superfluous parenthesis:

```suggestion
        if (iaaddr != NULL) {
```